packages/flutter/lib/src/painting/basic_types.dart

```diff
@@ -167,33 +167,134 @@ enum VerticalDirection {
   down,
 }
 
-/// A direction along either the horizontal or vertical [Axis].
+/// A direction along either the horizontal or vertical [Axis] in which the
+/// origin, or zero position, is determined.
+///
+/// This value relates to the direction in which the scroll offset increases
+/// from the origin. This value does not represent the direction of user input
+/// that may be modifying the scroll offset, such as from a drag. For the
+/// active scrolling direction, see [ScrollDirection].
+///
+/// {@tool dartpad}
+/// This sample shows a [CustomScrollView], with [Radio] buttons in the
+/// [AppBar.bottom] that change the [AxisDirection] to illustrate different
+/// configurations.
+///
+/// ** See code in examples/api/lib/painting/axis_direction/axis_direction.0.dart **
+/// {@end-tool}
+///
+/// See also:
+///
+///   * [ScrollDirection], the direction of active scrolling, relative to the positive
+///     scroll offset axis given by an [AxisDirection] and a [GrowthDirection].
+///   * [GrowthDirection], the direction in which slivers and their content are
+///     ordered, relative to the scroll offset axis as specified by
+///     [AxisDirection].
+///   * [CustomScrollView.anchor], the relative position of the zero scroll
+///     offset in a viewport and inflection point for [AxisDirection]s of the
+///     same cardinal [Axis].
+///   * [axisDirectionIsReversed], which returns whether traveling along the
+///     given axis direction visits coordinates along that axis in numerically
+///     decreasing order.
 enum AxisDirection {
-  /// Zero is at the bottom and positive values are above it: `⇈`
+  /// A direction in the [Axis.vertical] where zero is at the bottom and
+  /// positive values are above it: `⇈`
   ///
   /// Alphabetical content with a [GrowthDirection.forward] would have the A at
-  /// the bottom and the Z at the top. This is an unusual configuration.
+  /// the bottom and the Z at the top.
+  ///
+  /// For example, the default behavior of a [ListView] with [ListView.reverse]
+  /// set to true would have this axis direction.
+  ///
+  /// {@tool dartpad}
```

Comment:
I am actually wondering if dartdoc works well with these kind of large samples on an enum value since I believe enum values don't get their own dedicated page?

Reply:
I will check locally. I did not realize they don't have their own page. The individual axis direction samples are likely not necessary then

Reply:
Removed the individual samples. I had written them first before making the toggle-able one. I think that one is more comprehensive anyways.